src/validation.cpp

```diff
@@ -2246,17 +2247,30 @@ bool CChainState::FlushStateToDisk(
     {
         bool fFlushForPrune = false;
         bool fDoFullFlush = false;
+
         CoinsCacheSizeState cache_state = GetCoinsCacheSizeState(&m_mempool);
         LOCK(cs_LastBlockFile);
         if (fPruneMode && (fCheckForPruning || nManualPruneHeight > 0) && !fReindex) {
+            // make sure we don't prune below the blockfilterindexes bestblocks
+            // pruning is height-based
+            int index_lowest_blockheight = m_chain.Height();
+            bool has_blockfilter = false;
+            ForEachBlockFilterIndex([&index_lowest_blockheight, &has_blockfilter](BlockFilterIndex& index) {
+                int height = index.GetSummary().best_block_height;
+                if (height < index_lowest_blockheight) {
+                    index_lowest_blockheight = height;
+                }
+                has_blockfilter = true;
+            });
```

Comment:
Using `last_prune` instead of `m_chain.Height()` will lead to preserve 288 blocks below the artificial tip (`last_prune`) which IMO are not subject to reorg.

Reply:
> Using `last_prune` instead of `m_chain.Height()` will lead to preserve 288 blocks below the artificial tip (`last_prune`) which IMO are not subject to reorg.

If you're talking about the non-manual case, the you need to FindFilesToPrune a new parameter, not use the existing one, see https://github.com/bitcoin/bitcoin/pull/15946#discussion_r573391858. Not suggesting preserving below the tip.

Reply:
Yes. You're right. Lets just accept the 288blocks buffer for now (can be changed later if necessary).